lib/src/features/authenticate/presentation/bloc/authentication_bloc.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:bloc/bloc.dart';
+import 'package:dome_smart_home_app/src/features/authenticate/services/api/authentication_service.dart';
+
+import 'authentication_event.dart';
+import 'authentication_state.dart';
+
+class AuthenticationBloc
+    extends Bloc<AuthenticationEvent, AuthenticationState> {
+  AuthenticationBloc({required AuthenticationService authenticationService}) : super(AuthenticationInitial()) {
+    on<RegisterButtonPressed>((event, emit) {
+      emit(AuthenticationRegister());
+    });
+    on<LoginButtonPressed>((event, emit) {
```

Comment:
if 2 events are serving the same action (in this case you just emit the same state) we should just make a single event (e.g. AuthenticateUserEvent)